applications/nsmrs/pkg/serviceregistryserver/cache.go

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) 2019 Cisco and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package serviceregistryserver
+
+import (
+	"sync"
+	"time"
+
+	"github.com/networkservicemesh/networkservicemesh/utils"
+
+	"github.com/pkg/errors"
+
+	"github.com/golang/protobuf/ptypes/timestamp"
+	"github.com/sirupsen/logrus"
+
+	"github.com/golang/protobuf/proto"
+
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/registry"
+)
+
+const (
+	// NSEExpirationTimeoutDefault - default Endpoint expiration timeout, NSE will be deleted if UpdateNetworkServiceEndpoint not received
+	NSEExpirationTimeoutDefault = 5 * time.Minute
+	// NSEExpirationTimeoutEnv - environment variable contains custom NSEExpirationTimeout
+	NSEExpirationTimeoutEnv = utils.EnvVar("NSE_EXPIRATION_TIMEOUT")
+)
+
+// NSERegistryCache - cache of registered Network Service Endpoints
+type NSERegistryCache interface {
+	AddNetworkServiceEndpoint(nse *registry.NSERegistration) (*registry.NSERegistration, error)
+	UpdateNetworkServiceEndpoint(nse *registry.NSERegistration) (*registry.NSERegistration, error)
+	DeleteNetworkServiceEndpoint(endpointName string) (*registry.NSERegistration, error)
+	GetEndpointsByNs(networkServiceName string) []*registry.NSERegistration
```

Comment:
May be name this method as GetEndpoints? Why we need ByNs if we have just one method.